### PR TITLE
Mark openapi attributes as required for reads

### LINKED ISSUE #1744: [Feature Request] Mark nonnullable attributes as required in RESTApiHandler get requests
When using the RESTApiHandler and OpenAPI plugin, the return types for attributes in a get request are currently all optional (`<type> | undefined`). Since JSON:API responses always include all the attributes, they should not be marked as optional. This can be achieved with listing the attributes in a `required` array in the openapi schema.

Basically what we want is for the attributes to go from e.g.

**Before**
```ts
{
  myString: string | undefined
  myNullableString: string | null | undefined
}
```

**After**
```ts
{
  myString: string
  myNullableString: string | null
}
```

See [Discord discussion](https://discordapp.com/channels/1035538056146595961/1287697883906179105/1287697883906179105) for more details.

This change does not apply to RPCApiHandler since it allows selecting which attributes to return, so they're all optional in practice.

### PR DESCRIPTION
**Pull request by thomassnielsen**
Fixes #1744 by marking all fields as required for read requests. This allows for more accurate documentation and better type safety for generated clients, until sparse fieldsets are implemented.

### CODE REVIEW

**Comment by thomassnielsen** [comment]
As far as I can tell the test fails aren't related to my changes here(?)

**Comment by thomassnielsen** [comment]
I've only been running the tests within the openapi plugin folder and they succeed locally. I don't have enough memory to comfortably run the entire test suite locally.

File: packages/plugins/openapi/src/rest-generator.ts [hunk 1]
```diff
@@ -874,6 +874,9 @@ export class RESTfulOpenAPIGenerator extends OpenAPIGeneratorBase {
                     !(isDataModel(field.$resolvedType?.decl) && field.type.array)
                 ) {
                     required.push(field.name);
+                } else if (mode === 'read') {
+                    // Until we support sparse fieldsets, all fields are required for read operations
+                    required.push(field.name);
```
[the remark this answers is not in the document]
Comment (thomassnielsen) [reply packages/plugins/openapi/src/rest-generator.ts:879]: Not relevant for now

**Comment by ymc9** [comment]
> I've only been running the tests within the openapi plugin folder and they succeed locally. I don't have enough memory to comfortably run the entire test suite locally.

Hey it's caused by a typing issue with new release of tanstack-query. Test cases previously didn't pin to a specific dependency version. I've made a fix and merged it into your branch.

**Review by ymc9 — APPROVED** [review]

**Pull request merged**